go/vt/vtctl/workflow/stream_migrator.go

```diff
@@ -753,8 +813,17 @@ func (sm *StreamMigrator) syncSourceStreams(ctx context.Context) (map[string]rep
 					allErrors.RecordError(err)
 					return
 				}
-
-				query := fmt.Sprintf("update _vt.vreplication set state='Running', stop_pos='%s', message='synchronizing for cutover' where id=%d", replication.EncodePosition(pos), vrs.ID)
+				comment := ""
+				if vrs.WorkflowType == binlogdatapb.VReplicationWorkflowType_Materialize && vrs.BinlogSource.Keyspace == sm.ts.TargetKeyspaceName() {
+					// For intra-keyspace materializations in a keyspace that's being
+					// resharded, we don't have serving tablets in the source keyspace.
+					// So we instruct the VReplication engine and controller on the
+					// target tablets to include non-serving tablets in their search
+					// for source tablets to stream from.
```

Comment:
Why are the source tablets not-serving? Is it because we have already stopped writes on them?

Reply:
Which side / shard sets are serving is the primary thing we change when switching traffic -- there's always one side that's serving.  I'll update the comment to be more clear here. The source used here is source for the intra-keyspace materializations, whereas in the workflow these are on the tablets that we're about to switch traffic TO, but we have not made them serving yet as we're not past the point of no-return.